modules/ecs6-class/line.js

```diff
@@ -2,17 +2,29 @@ const Point = require("./point");
 
 class Line {
     constructor({ point1 = new Point(), point2 = new Point(), n = undefined, slope = undefined }) {
+        if(typeof(point1)!="object" && point1 !=undefined || typeof(point2)!="object" && point2 !=undefined){
+            throw new Error('argument must be type point')
+        }
+        if (typeof(n)!="number"&& n !=undefined|| typeof(slope)!="number" && slope !=undefined){
+            throw new Error('argument must be type number')
+        }
         this.point1 = point1;
         this.point2 = point2;
         this.slope = slope;
         this.n = n;
     }
 
     calculateSlope = () => {
+        if( this.point1.x==this.point1.y && this.point1.y==0 || this.point2.x==this.point2.y &&  this.point2.y==0){
+            throw new Error("missing data")
```

Comment:
it's not the correct error description

Reply:
this.slope = 0 ;

Reply:
throw new Error("missing data")dthrow new Error("attribute is undefine")

Reply:
throw new Error("attribute is undefine")